bayesian_cnn_prometheus/learning/bayesian_detector.py

```diff
@@ -114,8 +111,4 @@ def get_input_shape(dataset: tf.data.Dataset) -> Tuple[int, ...]:
         :param dataset: tensorflow dataset object
         :return: input shape
         """
-        return tuple([int(dim.value) for dim in list(dataset.element_spec[0].shape)[1:]])
-
-    @staticmethod
-    def _calculate_train_len():
-        return 100  # TODO How can we calc train len?
+        return tuple(list(dataset.element_spec[0].shape)[1:])
```

Comment:
To jakoś tak dziwnie tutaj widać, ale usunęłam funkcję _calculate_train_len (była potrzebna jedynie do tej starej funkcji lossu) i zaktualizowałam get_input_shape (z tf2 łatwiej to odczytać)

Reply:
Tak swoją drogą, to czemu tutaj `input_shape` jest `[1:]`, a np. [tutaj](https://github1s.com/aI-lab-glider/3d-cnn-prometheus/blob/HEAD/bayesian_cnn_prometheus/evaluation/bayesian_model_evaluator.py)  `[:3]`?
![image](https://user-images.githubusercontent.com/21079319/143948448-10f5bb80-8710-4573-a180-927f7e8ec1c8.png)


Mam wrażenie, że w różne momenty pomijamy różne wymiary.

Reply:
Tutaj chcemy wziąć input shape od pierwszego elementu, ponieważ element zerowy to wielkość naszego batcha (wielkość input shape bierzemy tutaj z naszego datasetu (generatora), dla którego pojedyncza wygenerowana instancja ma shape np. [32 (batch_size), 64 (width), 64 (height), 32 (depth), 1]. 
Na screenie jest [:3] od tego input shape, ponieważ chcemy wyciągnąć sobie z tego [32 (batch_size), 64 (width), 64 (height), 32 (depth)] czyli chunk_size w 3D. 
Ta jedynka na końcu input_shape jest wykorzystywana przy budowaniu architektury sieci. W sensie wykorzystywany jest cały input_shape [32 (batch_size), 64 (width), 64 (height), 32 (depth), 1], który jest funkcją Input() z TF przekształcany do [None, 32 (batch_size), 64 (width), 64 (height), 32 (depth), 1], żebyśmy mogli trenować na batchach o jakim chcemy rozmiarze. 
Nie do końca rozumiem, dlaczego potrzebujemy tej jedynki na końcu, tworzy ona dodatkowy wymiar, który prawdopododobnie jest wykorzystywany do przeształceń przy treningu modelu. Twórcy repozytorium, którym się inspirujemy mieli jako domyślny input również cztery wymiary [width, height, depth, 1]. Jest o tym dyskusja w https://trello.com/c/JbYlUn17

Reply:
Rozumie. 
Dzięki za tłumaczenie :)